.changeset/tender-forks-trade.md

```diff
@@ -0,0 +1,8 @@
+---
+"mobx": patch
+---
+
+-   by default all functions are converted to `autoAction`s/`flow`s
+-   by default all originally _own_ props are now observable and enumerable (as in pre v6)
+-   by default _lone_ setters are converted to `action`s
+-   `deep: false` ignores _all_ property values (including functions that would be previously converted to `autoAction`/`flow`)
```

Comment:
... so actions will have to be marked explicitly as (auto) `action`

Reply:
Yes, it really means do not touch the values.
However `deep: false` is to create shallow observable "container" - imo it doesn't make much sense to use `deep: false` for "smart" objects with actions etc, not sure what would be a good motivation for doing so.
Also note `deep: false` won't affect methods on prototype (even when bound) - it affects only `observable` (own) props.
It is possible to change this - I can move the logic from enhancer to annotation, therefore conversion to `autoAction` would only occur for objects - but then you have the opposite problem - you cannot create an object that is truly "shallow" - which seems wrong - and if I am not mistaken, this is currently the case as well - `deep: false` has no effect on `autoAction` inferrence, so even if user creates *shallow* object, we will mess with it's content (if the content is function) - and I am not even sure if we handle this consistently, the behavior may actually differ for initialization and property created via assigment, not sure.